cpp/src/examples/ucc_example.cpp

```diff
@@ -0,0 +1,179 @@
+/*
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <mpi.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <ucc/api/ucc.h>
+#include <iostream>
+
+#define STR(x) #x
+#define UCC_CHECK(_call)                                            \
+    if (UCC_OK != (_call)) {                                        \
+        fprintf(stderr, "*** UCC TEST FAIL: %s\n", STR(_call));     \
+        MPI_Abort(MPI_COMM_WORLD, -1);                              \
+    }
+
+static ucc_status_t oob_allgather(void *sbuf, void *rbuf, size_t msglen,
+                                  void *coll_info, void **req) {
```

Comment:
Are we going to C style for method naming or should we go for Camel case?

Reply:
This is just an example. We can remove this once UCC is properly integrated.